dial.go

```diff
@@ -40,72 +34,14 @@ func Dial(ctx context.Context, instance string) (net.Conn, error) {
 	if err != nil {
 		return nil, err
 	}
-	return d.dial(ctx, instance)
+	return d.Dial(ctx, instance)
 }
 
 // defaultDialer provides the singleton dialer as a default for dial functions.
-func defaultDialer() (*dialManager, error) {
+func defaultDialer() (*Dialer, error) {
 	// TODO: Provide functionality for customizing/setting the default dialer
 	once.Do(func() {
-		dm, dErr = newDialManager()
+		dm, dErr = NewDialer(context.Background())
 	})
 	return dm, dErr
 }
-
```

Comment:
This is probably stating the obvious, but this code could have easily stayed in this file. Same with options.go below. Typically Go files are larger (since they don't need to map one to one to a class in Java, for example). Mostly it's a matter of convenience to the reader (looking at one file vs jumping between files). Ultimately this is a matter of preference and up to you. Just wanted to call it out.

Reply:
If you prefer, I can put them back in one file. I typically find large go files both difficult to navigate and work in, and prefer some organizational boundaries so it's easier to read stuff with proper context. 

These are a bit on the small side at the moment, but I expect them to grow a fair amount so I just split them now.

Reply:
Keeping them as is works for me. Just wanted to bring it up.